### PR TITLE
Whole word scanning

### LINKED ISSUE #889: Support option to tap to scan whole word
Currently yomitan starts the word search at the location of the cursor, which makes it inconvenient for languages with spaces like Spanish where often when we tap a word we don't really care about highlighting the subword but rather the whole word. This is really easy to fat finger and look up a substring that we don't care about

Example:
image

The request is to add a yomitan option that scans the entire word independent of whatever letter within a word is tapped.

There are edge cases though, the biggest one I can think of is that with languages that have spaces, there are often definition entries that have spaces as well. For example, "get out" might have a separate dictionary entry, and I would want the behavior of yomitan to be able to reach the entry for "get out" no matter where within the word I scan.

### PR DESCRIPTION
**Pull request by jamesmaa**
Fixes #889 

- [x] : Popup flickers when moving within characters in a word, probably because yomitan considers the lookups to be different
- [x] : Doesn't work on google.com apparently according to @Casheeew

### CODE REVIEW

File: ext/js/background/backend.js [hunk 1]
```diff
@@ -900,6 +900,7 @@ export class Backend {
 
     /** @type {import('api').ApiHandler<'isTextLookupWorthy'>} */
     _onApiIsTextLookupWorthy({text, language}) {
+        if (text === null) { return false; }
```
Comment (jamesmaa) [inline ext/js/background/backend.js:903]: Idk why I had to add this otherwise there are errors

File: ext/js/dom/text-source-generator.js
Comment (jamesmaa) [inline ext/js/dom/text-source-generator.js:1]: Irrelevant typo changes here.

File: types/ext/dom-text-scanner.d.ts
Comment (jamesmaa) [inline types/ext/dom-text-scanner.d.ts:1]: This formatting of comments allows me to show the jsdoc comments when I hover over `ElementSeekInfo` variables. Was helpful for me to understand the code and I suggest we do more of these in the future.

File: ext/js/background/backend.js (hunk 1 above)
Comment (jamesmaa) [reply ext/js/background/backend.js]: okay it's gone after changes. idk what the original issue was. removing...

**Review by Kuuuube — COMMENTED** [review]
Is there any upper bound for how long a word is/how far backwards it can seek? I'm not seeing anything and this could lead to quite a lot of trouble if a user tries to scan an enormous string. Even a huge limit like 1000 characters would probably be a good idea.

**Comment by jamesmaa** [comment]
@Kuuuube it uses `this._scanLength` as the max limit for looking back. Basically it tells yomitan to try to set the startOffset back by `this._scanLength` characters and indicates for it to stop going back if it sees a whitespace. 

Reference
https://github.com/themoeway/yomitan/pull/1330/files#diff-255481c4f025ab785ed8746ef13607c1fcf4ad6fc4e7ee7aecd69a25244bf992R474

**Review by Kuuuube — CHANGES_REQUESTED** [review]
Looks good, will approve when tests are fixed. `test/options-util.test.js` needs the version bumped from 50 to 51.

**Review by Kuuuube — APPROVED** [review]

**Comment by Kuuuube** [comment]
Might want to have it pull the sentence termination characters and break on any of them. Though there is also the issue of should it stop on `'` because some words contain `'` but it could also be used to quote things.

**Comment by Casheeew** [comment]
> Probably seeking back into other HTML elements and concatenating. Layout-aware scanning doesn't seem to impact it. 

Might be the same or similar issue to what i demonstrated on discord.

**Comment by Casheeew** [comment]
> Might want to have it pull the sentence termination characters and break on any of them. Though there is also the issue of should it stop on `'` because some words contain `'` but it could also be used to quote things.

I think it's safe to just look one character before the quote and break if it's a whitespace, and not otherwise.

**Comment by Casheeew** [comment]
It now works on double quoted words but not on single quoted words.

**Comment by Casheeew** [comment]
Otherwise lgtm

**Comment by jamesmaa** [comment]
Okay done I followed your advice and peeked at the character before the single quotes to see if it's an apostrophe or wrapped in single quotes

**Review by Casheeew — APPROVED** [review]

**Pull request merged**